### PR TITLE
Make v0.6.2 latest released version in docs and deploy

### PR DESCRIPTION
**Pull request by davidz627**
/kind cleanup
/assign @msau42 

```release-note
NONE
```

### CODE REVIEW

File: deploy/kubernetes/overlays/prow-gke-release-staging-rc/kustomization.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ bases:
 images:
 - name: gke.gcr.io/gcp-compute-persistent-disk-csi-driver
   newName: gcr.io/gke-release-staging/gcp-compute-persistent-disk-csi-driver
-  newTag: "v0.6.1"
+  newTag: "v0.6.2"
```
Comment (msau42) [inline deploy/kubernetes/overlays/prow-gke-release-staging-rc/kustomization.yaml]: Should this be -gke.0?

Comment (davidz627) [reply deploy/kubernetes/overlays/prow-gke-release-staging-rc/kustomization.yaml]: it can be either - the staging gcr has both.

Comment (davidz627) [reply deploy/kubernetes/overlays/prow-gke-release-staging-rc/kustomization.yaml]: updated

**Pull request merged**